src/App.tsx

```diff
@@ -17,18 +70,39 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                searchTerm={searchTerm}
+                selectedStatus={selectedStatus}
+                onSearchChange={handleSearchChange}
+                onStatusChange={handleStatusChange}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {loading ? (
+                <Loader />
+              ) : (
+                <TodoList
+                  todos={todos.filter(todo => {
+                    const matchesTitle = todo.title
+                      .toLowerCase()
+                      .includes(searchTerm.toLowerCase());
+                    const matchesStatus =
+                      selectedStatus === TodoStatus.ALL ||
+                      (selectedStatus === TodoStatus.ACTIVE && !todo.completed) ||
+                      (selectedStatus === TodoStatus.COMPLETED && todo.completed);
+                    return matchesTitle && matchesStatus;
+                  })}
```

Comment:
Move this logic from props to the helper variable and use it here
```suggestion
                  todos={visibleTodos}
```